pkg/apis/operator/v1alpha1/apiserver_types.go

```diff
@@ -26,12 +26,21 @@ type APIServerSpec struct {
 	// TLSSecretRef references the TLS certificate and private key for serving the KubeCarrier API.
 	// +optional
 	TLSSecretRef *ObjectReference `json:"tlsSecretRef,omitempty"`
+	// +optional
+	// Authentication configuration
+	Authentication *Authentication `json:"authentication,omitempty"`
+}
+
+type Authentication struct {
 	// OIDC specifies OpenID Connect configuration for API Server authentication
 	// +optional
 	OIDC *APIServerOIDCConfig `json:"oidc,omitempty"`
 	// StaticUsers specifies static users configuration for API Server authentication
 	// +optional
 	StaticUsers *StaticUsers `json:"staticUsers,omitempty"`
+	// Order in which auth providers executed. Available values: Anonymous, Htpasswd, OIDC, Token
+	// +optional
+	Order []string `json:"order,omitempty"`
```

Comment:
If we are going to use `Order` list, we need to add a webhook to check:
1. `OIDC` != nil if OIDC is in `Order` list
2. `StaticUser` != nil if Htpasswd is in `Order` list

Reply:
Also, I think the `Order` list should not be `omitempty`? what will happen in your case if the `Order` is empty?

Reply:
> Also, I think the `Order` list should not be `omitempty`? what will happen in your case if the `Order` is empty?

I will add `Anonymous` by default

> If we are going to use `Order` list, we need to add a webhook to check:
> 
> 1. `OIDC` != nil if OIDC is in `Order` list
> 2. `StaticUser` != nil if Htpasswd is in `Order` list

we don't have webhooks for apiserver as for now, and I don't think it worth adding it for that, what we can easily do, if config is `nil` I can delete this provider from `Order`

Reply:
No, we only need to add logic to operator webhook: https://github.com/kubermatic/kubecarrier/blob/master/pkg/operator/internal/webhooks/kubecarrier_webhook.go, since apiserver.spec is enabled in kubecarrier

Reply:
oh, ok, will do that